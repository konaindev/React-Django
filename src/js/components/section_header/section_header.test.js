import renderer from "react-test-renderer";
import { SectionHeader } from "./index";

const props = {
  title: "Test Section"
};

describe("SectionHeader", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SectionHeader {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders default mode", () => {
    let wrapper = shallow(<SectionHeader {...props} />);
    const titleEl = wrapper.find("p.section-header__title");
    expect(titleEl).toHaveText(props.title);
  });

  it("renders with small margin", () => {
    let wrapper = shallow(<SectionHeader {...props} smallMarginTop />);
    expect(wrapper).toHaveClassName("section-header--mt-sm");
  });

  it("renders with right-side content", () => {
    const rightSideText = "Right-Side Content";
    let wrapper = shallow(
      <SectionHeader {...props}>{rightSideText}</SectionHeader>
    );
    const extraEl = wrapper.find(".section-header__extra");
    expect(extraEl).toHaveLength(1);
    expect(extraEl).toHaveText(rightSideText);
  });
});
