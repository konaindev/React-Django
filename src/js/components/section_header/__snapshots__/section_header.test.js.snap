// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader renders correctly 1`] = `
<div
  className="section-header"
>
  <p
    className="section-header__title"
  >
    Test Section
  </p>
</div>
`;
